CRM/PivotData/DataContact.php

```diff
@@ -0,0 +1,207 @@
+      );
+
+      while ($customFieldsResult->fetch()) {
+        $customField = new CRM_Core_BAO_CustomField();
+        $customField->id = $customFieldsResult->id;
+        $customField->find(true);
+
+        $fields['contact']['custom_' . $customFieldsResult->id] = array(
+          'name' => 'custom_' . $customFieldsResult->id,
+          'title' => $customFieldsResult->label,
+          'pseudoconstant' => array(
+            'optionGroupName' => $customFieldsResult->option_group_name,
+          ),
+          'customField' => (array)$customField,
+        );
+      }
+      //Add contact fields
+      $fields['contact']['display_name'] = ['name' => 'display_name', 'title' => 'Display Name'];
+      $fields['contact']['contact_type'] = ['name' => 'contact_type', 'title' => 'Contact Type'];
+      $fields['contact']['contact_id'] = ['name' => 'contact_id', 'title' => 'Contact ID'];
+      $fields['contact']['birth_date'] = ['name' => 'contact_id', 'title' => 'Birth Date'];
+
+      $fields['jobcontract']['id'] = ['name' => 'id', 'title' => ts('Contract ID')];
+      $fields['jobcontract']['position'] = ['name' => 'position', 'title' => ts('Contract Position')];
+      $fields['jobcontract']['title'] = ['name' => 'title', 'title' => ts('Contract Title')];
+      $fields['jobcontract']['contract_type'] = ['name' => 'contract_type', 'title' => ts('Contract Type')];
+      $fields['jobcontract']['period_start_date'] = ['name' => 'period_start_date', 'title' => ts('Contract Start Date')];
+      $fields['jobcontract']['period_end_date'] = ['name' => 'period_end_date', 'title' => ts('Contract End Date')];
+      $fields['jobcontract']['is_current'] = ['name' => 'is_current', 'title' => ts('Current Contract')];
+
+      $fields['jobroles']['id'] = ['name' => 'id', 'title' => ts('Role ID')];
+      $fields['jobroles']['start_date'] = ['name' => 'start_date', 'title' => ts('Role Start Date')];
+      $fields['jobroles']['end_date'] = ['name' => 'end_date', 'title' => ts('Role End Date')];
+      $fields['jobroles']['title'] = ['name' => 'title', 'title' => ts('Role Title')];
+      $fields['jobroles']['description'] = ['name' => 'description', 'title' => ts('Role Description')];
+      $fields['jobroles']['location'] = ['name' => 'location', 'title' => ts('Role Location')];
+      $fields['jobroles']['department'] = ['name' => 'department', 'title' => ts('Role Department')];
+      $fields['jobroles']['functional_area'] = ['name' => 'functional_area', 'title' => ts('Role Functional Area')];
+      $fields['jobroles']['hours'] = ['name' => 'hours', 'title' => ts('Role Hours')];
+      $fields['jobroles']['role_hours_unit'] = ['name' => 'role_hours_unit', 'title' => ts('Role Hours Unit')];
+      $fields['jobroles']['level_type'] = ['name' => 'level_type', 'title' => ts('Role Level Type')];
+      $fields['jobroles']['organization'] = ['name' => 'organization', 'title' => ts('Role Organization')];
```

Comment:
Quite a manual job here. Do we really need to manually create this list of fields? Also, there's some repetition in the field name, as it is used as the key and as the name. I believe this is just the format that the extension expects the data to be?

Reply:
Yes, its just the format that the extension expects the data to be, a way to automate might be to fetch the DAO fields for each of the entity and then unset the fields we don't want or set the ones we want. But most of the column title needs to be modified to suit what would be in the report which most are different in the BAO which is why I prefer the manual way.